packages/ckeditor5-ckbox/tests/ckboximageedit/ckboximageeditcommand.js

```diff
@@ -372,6 +373,79 @@ describe( 'CKBoxImageEditCommand', () => {
 				sinon.assert.callCount( respondSpy, 4 );
 			} );
 
+			it( 'should add a pending action after a change and remove after server response', done => {
+				const pendingActions = editor.plugins.get( PendingActions );
+				setModelData( model, '[<imageBlock alt="alt text" ckboxImageId="example-id" src="/assets/sample.png"></imageBlock>]' );
+				const clock = sinon.useFakeTimers();
+
+				sinonXHR.respondWith( 'GET', CKBOX_API_URL + '/assets/image-id1', [
+					200,
+					{ 'Content-Type': 'application/json' },
+					JSON.stringify( {
+						metadata: {
+							metadataProcessingStatus: 'success'
+						}
+					} )
+				] );
+
+				const dataMock = {
+					data: {
+						id: 'image-id1',
+						extension: 'png',
+						metadata: {
+							width: 100,
+							height: 100
+						},
+						name: 'image1',
+						imageUrls: {
+							100: 'https://example.com/workspace1/assets/image-id1/images/100.webp',
+							default: 'https://example.com/workspace1/assets/image-id1/images/100.png'
+						},
+						url: 'https://example.com/workspace1/assets/image-id1/file'
+					}
+				};
+
+				const dataMock2 = {
+					data: {
+						id: 'image-id2',
+						extension: 'png',
+						metadata: {
+							width: 100,
+							height: 100
+						},
+						name: 'image2',
+						imageUrls: {
+							100: 'https://example.com/workspace1/assets/image-id2/images/100.webp',
+							default: 'https://example.com/workspace1/assets/image-id2/images/100.png'
+						},
+						url: 'https://example.com/workspace1/assets/image-id2/file'
+					}
+				};
+
+				command.on( 'ckboxImageEditor:processed', () => {
+					expect( pendingActions.hasAny ).to.be.false;
+					expect( pendingActions._actions.length ).to.equal( 0 );
```

Comment:
How does it work? We should have two `'ckboxImageEditor:processed'` events and after the first one there should be `1` pending action.
I suspect, it is because both `onSave` are called at the same time. I would add `clock.tickAsync` in between them.